frontend/public/components/debug-terminal.tsx

```diff
@@ -0,0 +1,235 @@
+    },
+  };
+};
+
+const DebugTerminalError: React.FC<DebugTerminalErrorProps> = ({ error }) => {
+  return (
+    <div className="co-m-pane__body">
+      <Alert variant="danger" isInline title={error} />
+    </div>
+  );
+};
+
+const DebugTerminalInner: React.FC<DebugTerminalInnerProps> = ({ obj, activeContainer }) => {
+  const { t } = useTranslation();
+  const infoMessage = (
+    <div className="co-terminal-info-message">
+      <Alert
+        variant="info"
+        isInline
+        title={t(
+          'public~This temporary pod has a modified entrypoint command to debug a failing container. The pod will be available for one hour and will be deleted when the terminal window is closed.',
+        )}
+      />
+    </div>
+  );
+  switch (obj?.data?.status?.phase) {
+    case 'Failed':
+      return (
+        <DebugTerminalError
+          error={
+            <>
+              {t('public~The debug pod failed. ')}
+              {obj?.data?.status?.containerStatuses?.[0]?.state?.terminated?.message ||
+                obj?.data?.status?.message}
+            </>
+          }
+        />
+      );
+    case 'Running':
+      return (
+        <PodExecLoader obj={obj.data} activeContainer={activeContainer} infoMessage={infoMessage} />
+      );
+    default:
+      return <LoadingBox />;
+  }
+};
+
+export const DebugTerminal: React.FC<DebugTerminalProps> = ({
+  obj,
+  match: {
+    params: { name },
+  },
+}) => {
+  const [resources, setResources] = React.useState<FirehoseResource[]>([]);
+  const [errorMessage, setErrorMessage] = React.useState('');
+  const podName = obj?.metadata?.name;
+  const podNamespace = obj?.metadata?.namespace;
+  const containerName = name ? name : obj?.spec?.containers?.[0].name;
+  React.useEffect(() => {
+    const debugPodName = `${podName}-${containerName}-debug`;
+    const deleteDebugPod = async (podToDelete) => {
+      try {
+        await k8sKillByName(PodModel, podToDelete, podNamespace);
+      } catch (e) {
+        // eslint-disable-next-line no-console
+        console.warn('Could not delete container terminal debug pod.', e);
+      }
+    };
+    const createDebugPod = async () => {
+      try {
+        const podToCreate = await getDebugPod(debugPodName, obj, containerName);
+        const debugPod = await k8sCreate(PodModel, podToCreate);
+        if (debugPod) {
+          setResources([
+            {
+              isList: false,
+              kind: 'Pod',
+              name: debugPodName,
+              namespace: podNamespace,
+              prop: 'obj',
+            },
+          ]);
+        }
+      } catch (e) {
+        setErrorMessage(e.message);
+      }
+    };
+    createDebugPod();
+    window.addEventListener('beforeunload', deleteDebugPod);
+    return () => {
+      deleteDebugPod(debugPodName);
+      window.removeEventListener('beforeunload', deleteDebugPod);
+    };
+  }, [containerName, podName, podNamespace, obj]);
```

Comment:
Won't this create a new pod any time `obj` changes?

Reply:
Fixed